src/meta/src/manager/catalog/mod.rs

```diff
@@ -1743,6 +1743,34 @@ impl CatalogManager {
         Ok(version)
     }
 
+    pub async fn alter_table_owner(
+        &self,
+        table_id: SchemaId,
+        owner_id: UserId,
+    ) -> MetaResult<NotificationVersion> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+        database_core.ensure_table_id(table_id)?;
+
+        let mut table = database_core.tables.get(&table_id).unwrap().clone();
+        table.owner = owner_id;
+        // CREATE TABLE (...) WITH (...) OWNER xxx
+        // TODO: table.definition = `alter_owner_if...`
+
+        let version = self
```

Comment:
We need to persist it in meta store rather than just changing the owner in memory cache. Please wrap it with `BTreeMapTransaction` and handle it in this way:
```rust
let mut tables = BTreeMapTransaction::new(&mut database_core.tables);
let mut table = tables.get_mut(&table_id).unwrap();
table.owner = owner_id;
commit_meta!(self, tables)?;

// notify
```